File: android/erc681-sdk/src/main/kotlin/com/openpasskey/erc681/NativeAsset.kt
// SPDX-License-Identifier: Apache-2.0
// Copyright 2026 Victor Zhang

package com.openpasskey.erc681

/** OPK Protocol 1.6's EIP-7528 identifier for the current chain's native asset. */
object NativeAsset {
    const val SENTINEL = "0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE"
    const val DECIMALS = 18
    const val PROTOCOL_VERSION = "1.6"

    @JvmField
    val address: EvmAddress = EvmAddress.parse(SENTINEL)

    @JvmStatic
    fun isNative(address: EvmAddress): Boolean = address == this.address
}

File: ios/Sources/OPKTerminalCore/NativeAsset.swift
// SPDX-License-Identifier: Apache-2.0
// Copyright 2026 Victor Zhang

import Foundation

/// OPK Protocol 1.6's EIP-7528 identifier for the current chain's native asset.
public enum NativeAsset {
    public static let sentinelHex = "0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE"
    public static let address = try! EthereumAddress(hex: sentinelHex)
    public static let decimals: UInt8 = 18

    public static func isNative(_ address: EthereumAddress) -> Bool {
        address == self.address
    }
}

public extension PaymentToken {
    var isNativeAsset: Bool { NativeAsset.isNative(address) }
}
